confero/static/js/graph.js

```diff
@@ -0,0 +1,79 @@
+(function() {
+  const fetchData = type =>
+    fetch(`/api/graph/${type}`).then(response => response.json());
+
+  const PARTY_COLORS = {
+    DEM: "blue",
+    REP: "red",
+    IND: "yellow",
+    default: "black"
+  };
+  const nodeColor = candidate =>
+    PARTY_COLORS[candidate.party] || PARTY_COLORS.default;
+
+  const nodeSize = candidate =>
+    candidate.party === "IND" ? 20 : candidate.office === "P" ? 10 : 5;
+
+  const main = async () => {
+    const svg = d3.select("svg");
+
+    const width = 960;
+    const height = 600;
+
+    const [candidates, connections] = await Promise.all([
+      fetchData("candidates"),
+      fetchData("connections")
+    ]);
+
+    const simulation = d3
+      .forceSimulation(candidates)
+      .force("link", d3.forceLink(connections).id(node => node.id))
+      .force("charge", d3.forceManyBody().strength(() => -4))
+      .force("center", d3.forceCenter(width / 2, height / 2));
```

Comment:
There's a bunch of stuff in there, all of which i just copied from D3 tutorials. This is the meat though, that calculates the "force directed graph"